ui/cypress/e2e/explorer.test.ts

```diff
@@ -270,6 +270,39 @@ describe('DataExplorer', () => {
         cy.getByTestID('timerange-popover--dialog').should('have.length', 1)
       })
 
+      it('should not allow users to start dates after the stop date & stop dates before the start date', () => {
+        // Set the start date to Oct 29, 2019
+        cy.get('input[title="Start"]')
+          .should('have.length', 1)
+          .clear()
+          .type('2019-10-29 08:00:00.000')
+
+        // Set the stop date to Oct 29, 2019
+        cy.get('input[title="Stop"]')
+          .should('have.length', 1)
+          .clear()
+          .type('2019-10-29 09:00:00.000')
+
+        // click button and see if time range has been selected
+        cy.get('.cf-button--label')
+          .contains('Apply Time Range')
+          .should('have.length', 1)
+          .click()
+
+        cy.getByTestID('timerange-dropdown').click()
+        cy.getByTestID('dropdown-item-customtimerange').click()
+
+        // Select the 30th in the Start timerange
+        cy.get('.react-datepicker__day--030')
+          .first()
+          .should('be', 'disabled')
+
+        // Select the 28th in the Stop timerange
+        cy.get('.react-datepicker__day--028')
+          .last()
+          .should('be', 'disabled')
```

Comment:
This could probably be added to another test that selects the time range.  Also, a link to the bug would be 👍

Reply:
It's not a bug, more a weird UI quirk where we allow start times to be before the start time. It's associated with the issue in the PR comments, but that's a pretty wide-ranging issue.

#15632